### ISSUE TITLE
[Support]: retention 

### ISSUE THREAD

**Issue by heapxor** [blakeblackshear/frigate#2698]
### Describe the problem you are having

the documentation says:
Frigate chooses the largest matching retention value between the recording retention and the event retention when determining if a recording should be removed.

But I am not clear about that... how exactly can I set to retention of recordings, snapshots lets say for 1 year?

thanks!

### Version

latest

### Frigate config file

```yaml
no config at the moment
```

### Relevant log output

```shell
nothing
```

### FFprobe output from your camera

```shell
nothing
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

Other

### Network connection

Wired

### Camera make and model

hikvision

**Comment by blakeblackshear** [comment]
You will have to be more specific. If you want to retain all recordings and snapshots for a year, you would simply set retain_days to 365 for snapshots and recordings.

**Comment by heapxor** [comment]
@blakeblackshear not sure ... as per https://docs.frigate.video/configuration/index/
do i have to define it for each object ie events, snapshots 
or mode: all is enough?


# Optional: Record configuration
# NOTE: Can be overridden at the camera level
record:
  # Optional: Enable recording (default: shown below)
  # WARNING: Frigate does not currently support limiting recordings based
  #          on available disk space automatically. If using recordings,
  #          you must specify retention settings for a number of days that
  #          will fit within the available disk space of your drive or Frigate
  #          will crash.
  enabled: True
  # Optional: Number of minutes to wait between cleanup runs (default: shown below)
  # This can be used to reduce the frequency of deleting recording segments from disk if you want to minimize i/o
  expire_interval: 60
  # Optional: Retention settings for recording
  retain:
    # Optional: Number of days to retain recordings regardless of events (default: shown below)
    # NOTE: This should be set to 0 and retention should be defined in events section below
    #       if you only want to retain recordings of events.
    days: 180
    # Optional: Mode for retention. Available options are: all, motion, and active_objects
    #   all - save all recording segments regardless of activity
    #   motion - save all recordings segments with any detected motion
    #   active_objects - save all recording segments with active/moving objects
    # NOTE: this mode only applies when the days setting above is greater than 0
    mode: all


thanks

**Comment by NickM-27** [comment]
@heapxor You would just need to define 

```yaml
record:
  events:
    default: 365
```

to keep the recordings for a year.

The modes (all, motion, active_objects) are explained here: https://docs.frigate.video/configuration/record#what-do-the-different-retain-modes-mean

**Comment by heapxor** [comment]
@NickM-27 so confusing ... not saying there is also object snapshots

**Comment by NickM-27** [comment]
> @NickM-27 so confusing ... not saying there is also object snapshots

I said for recordings, to also keep snapshots for 1 year:

```yaml
snapshots:
  retain:
    default: 365
```

**Comment by heapxor** [comment]
@NickM-27 well not straightforward from doc... and so confusing ... maybe for myself, but yes confusion.

**Comment by NickM-27** [comment]
@heapxor Okay, do you have it sorted out now or do you still have questions?

**Comment by heapxor** [comment]
@NickM-27 i set is as you said. so lets see.

event/video (mp4) equals to Download Clip?
![image](https://user-images.githubusercontent.com/64676/159595041-cb82f725-b4d3-4160-a24f-4169c4905eab.png)

**Comment by NickM-27** [comment]
> @NickM-27 i set is as you said. so lets see.
> 
> event/video (mp4) equals to Download Clip? ![image](https://user-images.githubusercontent.com/64676/159595041-cb82f725-b4d3-4160-a24f-4169c4905eab.png)

I'm not sure what you're asking, pressing download clip, assuming there is a recording clip available for that event, will download the mp4

**Comment by heapxor** [comment]
@NickM-27 i mean that in terms of objects in config files; if u want to be able to download Clip you have to enable inside the config Events (object), so once the frigate is creating Events objects then you are only able to download mp4 via Download Clip. Correct? thanks

**Comment by NickM-27** [comment]
> @NickM-27 i mean that in terms of objects in config files; if u want to be able to download Clip you have to enable inside the config Events (object), so once the frigate is creating Events objects then you are only able to download mp4 via Download Clip. Correct? thanks

Yes that is correct, with that config above new events will have an mp4 available for download
